frontend/components/utils/BidInput.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+    <div class="w-full inline-block relative flex-shrink-0">
+        <Tooltip :visible="!!error" placement="topLeft" :title="error" class="items-center">
+            <div v-if="!amountToBid" class="Overlay TotalPrice">
+                <format-currency v-if="totalPrice && !disabled" :value="totalPrice" />
+                <span v-else class="opacity-50">UNKNOWN</span>
+            </div>
+            <span class="Overlay right-1">DAI</span>
+            <Input
+                v-model="amountToBidInput"
+                :disabled="disabled"
+                class="Input"
+                :class="{ Error: error }"
+                @focus="hideTotalPrice()"
+                @blur="showTotalPriceIfEmpty()"
+            />
+        </Tooltip>
+    </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import BigNumber from 'bignumber.js';
+import { Input, Tooltip } from 'ant-design-vue';
+import FormatCurrency from './FormatCurrency.vue';
+export default Vue.extend({
+    components: { FormatCurrency, Tooltip, Input },
+    props: {
+        totalPrice: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: null,
+        },
+        minimumDepositDai: {
+            type: Object as Vue.PropType<BigNumber>,
+            required: true,
+        },
+        amountToBid: {
+            type: Object as Vue.PropType<BigNumber> | undefined,
+            default: undefined,
+        },
+        disabled: {
+            type: Boolean,
+            default: false,
+        },
+    },
+    data() {
+        return {
+            amountToBidInput: '' as string,
+        };
+    },
+    computed: {
+        isInputEmpty(): boolean {
+            return this.amountToBidInput.trim() === '';
+        },
+        amountToBidInputParsed(): BigNumber | undefined {
+            if (this.isInputEmpty) {
+                return undefined;
+            }
+            const amountToBid = new BigNumber(this.amountToBidInput);
+            return amountToBid;
+        },
+        error(): string | undefined {
+            const maximumBid = this.totalPrice?.minus(this.minimumDepositDai) || new BigNumber(0);
+            if (this.amountToBidInputParsed?.isGreaterThan(maximumBid)) {
+                return `The bidding amount can not be greater than ${maximumBid.toFixed(2)} DAI`;
+            }
+            if (this.amountToBidInputParsed?.isLessThan(this.minimumDepositDai)) {
+                return `The bidding amount can not be smaller than ${this.minimumDepositDai.toFixed(2)} DAI`;
+            }
+            return undefined;
```

Comment:
It's ok to be explicit (so you don't have to edit anything here). I just hope that you know that undefined is returned by a function if no return is specified, so this is technically the same:
```suggestion
            return;
```
or
```suggestion
```